src/wallet/wallet.cpp

```diff
@@ -4438,6 +4461,8 @@ util::Result<MigrationResult> MigrateLegacyToDescriptor(const std::string& walle
     // both before and after reloading. This ensures the set is complete even if one of the wallets
     // fails to reload.
     std::set<fs::path> wallet_dirs;
+    // Reset options.require_format as options are used by reload_wallet which may reload wallets of any format.
+    options.require_format = std::nullopt;
```

Comment:
In f732495191d346b8:

The reset call needs to be moved above the `reload_wallet` function declaration. Otherwise, the wallet will be reloaded in read-only mode for any failure occurring in-between the `reload_wallet` declaration and this line, such as failures during backup creation or the decryption procedure.

I have written a test to trigger this misbehavior: furszy@c24b673 (feel free to pull it here). The test passes when the fix is cherry-picked on top: furszy@2d5d9a4.

Reply:
Fixed and pulled the test. I decided to just move this into `reload_wallet` directly, it doesn't need to live outside of it.